### PR TITLE
[1.0] Allow for gatsby-remark-smartypants options

### PR DESCRIPTION
**Pull request by mitchejj**
By default gatsby-remark-smartypants (retext-smartypant) renders -- as em-dash's. 

The [canonical implementation](http://daringfireball.net/projects/smartypants/) states that -- (two dashes) rendered as an en-dash while --- (three dashes) rendered as an em-dash.

This pull request would  would allow for that option along with any other retext-smartypant option.

### CODE REVIEW

File: packages/gatsby-remark-smartypants/src/index.js [hunk 1]
```diff
@@ -1,10 +1,13 @@
 const retext = require(`retext`)
 const visit = require(`unist-util-visit`)
 const smartypants = require(`retext-smartypants`)
+const _ = require('lodash')
 
-module.exports = ({ markdownAST }) => {
+module.exports = ({ markdownAST pluginOptions }) => {
```
Comment (KyleAMathews) [inline packages/gatsby-remark-smartypants/src/index.js]: Missing the comma between args here.

Comment (KyleAMathews) [inline packages/gatsby-remark-smartypants/src/index.js]: And actually, I moved the cheese on you a bit. Everywhere else, pluginOptions is the second argument so I just add https://github.com/gatsbyjs/gatsby/pull/1167 to make remark plugins work the same way. So your code should now be `module.exports = ({ markdownAST }, pluginOptions }) => {`.

File: packages/gatsby-remark-smartypants/src/index.js [hunk 2]
```diff
@@ -1,10 +1,13 @@
...
-module.exports = ({ markdownAST }) => {
+module.exports = ({ markdownAST }, pluginOptions = {}) =>
   visit(markdownAST, `text`, node => {
-    const processedText = String(retext().use(smartypants).process(node.value))
+    const defaultOptions = {}
+    const options = _.defaults(pluginOptions, defaultOptions)
+    const processedText = String(retext().use(smartypants, options).process(node.value))
     node.value = processedText
   })
 }
```
Comment (KyleAMathews) [inline packages/gatsby-remark-smartypants/src/index.js]: this isn't needed anymore. It's good btw to run `npm run format` from the base of the repo on PRs as that'll fix formatting + catch syntax errors like this. Need to create a checklist for PRs :-)

**Comment by KyleAMathews** [comment]
Thanks for the great work! Great to have the underlying options exposed now.

**Comment by jlengstorf** [comment]
Hiya @mitchejj! 👋

This is definitely late, but on behalf of the entire Gatsby community, I wanted to say thank you for being here.

Gatsby is built by awesome people like you. Let us say “thanks” in two ways:

1.  **We’d like to send you some Gatsby swag.** As a token of our appreciation, you can go to the [Gatsby Swag Store][store] and log in with your GitHub account to get a coupon code good for one free piece of swag. (We’ve got t-shirts and hats, plus some socks that are really razzing our berries right now.)
2.  **If you’re not already part of it, we just invited you to join the Gatsby organization on GitHub.** This will add you to our team of maintainers. You’ll receive an email shortly asking you to confirm. By joining the team, you’ll be able to label issues, review pull requests, and merge approved pull requests.

If you have questions, please don’t hesitate to reach out to us: tweet at [@gatsbyjs][twitter] and we’ll come a-runnin’.

Thanks again! 💪💜

[store]: https://store.gatsbyjs.org
[twitter]: https://twitter.com/gatsbyjs

**Pull request merged**